google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ReadWriteTransaction.java

```diff
@@ -225,13 +237,22 @@ public boolean isReadOnly() {
   }
 
   @Override
-  void checkValidTransaction() {
+  void checkValidTransaction(ParsedStatement statement) {
```

Comment:
nit: `checkValidationTransaction` mutates state within the transaction (it did before as well with the future), so perhaps it is a good chance to rename the method? (very nitty, so feel free to ignore)

Reply:
That's indeed a good point. Renamed to `checkOrCreateValidTransaction`.